pricelist_item_generator/models/pricelist.py

```diff
@@ -0,0 +1,415 @@
+# coding: utf-8
+# © 2016 David BEAL @ Akretion
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+from openerp import fields, models, api, _
+from openerp.exceptions import Warning as UserError
+import openerp.addons.decimal_precision as dp
+from collections import defaultdict
+
+
+TODO_SELECTION = [('update', 'Update'), ('create', 'Create')]
+
+
+def update_item_tpl_or_condition(vals, self):
```

Comment:
This function is not inside a class, then that intentionnaly that I put self arg  in secondth position to avoid confusion with a method. api multi is for method in class.